python/cudf/dataframe/multiindex.py

```diff
@@ -115,12 +130,53 @@ def __repr__(self):
         return "MultiIndex(levels=" + str(self.levels) +\
                ",\ncodes=" + str(self.codes) + ")"
 
+    @property
+    def codes(self):
+        if self._codes is not None:
+            return self._codes
+        else:
+            self._compute_levels_and_codes()
+            return self._codes
+
+    @property
+    def levels(self):
+        if self._levels is not None:
+            return self._levels
+        else:
+            self._compute_levels_and_codes()
+            return self._levels
+
     @property
     def labels(self):
         warnings.warn("This feature is deprecated in pandas and will be"
                       "dropped from cudf as well.", FutureWarning)
         return self.codes
 
+    def _compute_levels_and_codes(self):
+        levels = []
+        from cudf import DataFrame
+        codes = DataFrame()
+        names = []
+        # Note: This is an O(N^2) solution using gpu masking
+        # to compute new codes for the MultiIndex. There may be
+        # a faster solution that could be executed on gpu at the same
+        # time the groupby is calculated.
+        for by in self._source_data.columns:
+            if len(self._source_data[by]) > 0:
+                level = self._source_data[by].unique()
+                replaced = self._source_data[by].replace(
+                        level, Series(range(len(level))))
```

Comment:
Would much rather fix this down in the `replace.pyx` but at least file an issue and comment it if possible 😄